src/Diffusion.jl

```diff
@@ -1,9 +1,76 @@
 module Diffusion
+using StaticArrays 
 
 export absorb_boundry
+export reflect_boundary 
+export slope
+export midpoint
+export circleintersect
+export rslope
+export perpslope
+export solve
+export posFinal
 
 function absorb_boundry(pos,r)
-    return pos + r
+    if(((pos[1]^2)+(pos[2]^2))^(1/2) < r)
+    return pos
+    else
+    return nothing
+    end
+end
+
+function slope(pos1,pos2)               #finds the slope between two points
+    return (pos1[2]-pos2[2])/(pos1[1]-pos2[1])
+end
+
+function midpoint(pos1,pos2)            #finds the midpoint between two points
+    x = (pos1[1]+pos2[1])/2
+    y = (pos1[2]+pos2[2])/2
+    return SA[x,y]
+ end
+
+function circleintersect(pos1,pos2,r)   #estimates the intersection between the line formed by two points and a circle of radius r
+    mp = midpoint(pos1,pos2)
+    if(isapprox(((pos1[1]^2)+(pos1[2]^2))^(1/2),r))
```

Comment:
Julia doesn't require you to put the conditional in parentheses, but it's fine to do so.